datastore/google/cloud/datastore/query.py

```diff
@@ -454,7 +455,7 @@ def _process_query_results(self, response_pb):
         """
         self._skipped_results = response_pb.batch.skipped_results
 
-        if response_pb.batch.end_cursor == b'':  # Empty-value for bytes.
+        if response_pb.batch.more_results == _NO_MORE_RESULTS:
```

Comment:
I'm fairly certain there **was** a reason we did this, i.e. the server API didn't quite behave the way we expected.

I'd prefer sign off from the backend team before making the change.

Reply:
Hi there. :smile: I'm on the Datastore backend team.

This change looks good to me, but I'm thinking the check should actually mirror L487, i.e. `response_pb.batch.more_results in _FINISHED` (see https://cloud.google.com/datastore/docs/reference/rpc/google.datastore.v1#google.datastore.v1.QueryResultBatch.MoreResultsType).

cc @pcostell

Reply:
Talked with @pcostell; the original proposal (`NO_MORE_RESULTS`) is correct. :thumbsup:

Reply:
Thanks @wsh!

Reply:
@dhermes could it be merged now? Also do you know when is the next release of `datastore` submodule planned? Thanks.